pot/pot.go

```diff
@@ -448,14 +448,36 @@ func union(t0, t1 *Pot, pof Pof) (*Pot, int) {
 	return n, common
 }
 
+// ValConsumer is a function that consumes a Val and returns if it wants to consume more or not
+// Consumer<Val> in generic notation
+type ValConsumer func(Val) bool
+
+// ValIterator is a function that iterates values and executes for each of them a supplied ValConsumer.
+// it returns the result of the last ValConsumer executed. Usually is the pin of a pot but if could be the last element
```

Comment:
> Usually is the pin of a pot

I am not sure I understand what you mean here.

Reply:
Almost all uses of this function pivotVal is k.base, so the node address and the "pin" of the pot. 
In that situation eachBin is just traversing the bins from less po to more po.

Reply:
Ok. But I am unsure about the wording here. "Usually the value passed to the ValConsumer is the pin"

If we want a tutorial-ish overview of the use cases in the comment, maybe they should be in bullet points and give a bit more context. What do you think?

Reply:
In fact I think I'm going to change it. The last val consumed will be always the pivot value.